optimizer/plan/range.go

```diff
@@ -272,7 +272,11 @@ func (r *rangeBuilder) buildFromPatternLike(x *ast.PatternLikeExpr) []rangePoint
 		r.err = ErrUnsupportedType.Gen("NOT LIKE is not supported.")
 		return fullRange
 	}
-	pattern := x.Pattern.GetValue().(string)
+	pattern, err := types.ToString(x.Pattern.GetValue())
+	if err != nil {
+		r.err = errors.Trace(err)
```

Comment:
Yes, the error occurs when the type of the value is unknown. In the case when we added a new type, but forgot to handle it in ToSting method.